apps/emqx_conf/src/emqx_conf_schema.erl

```diff
@@ -693,13 +689,21 @@ tr_cluster_discovery(Conf) ->
 tr_logger_level(Conf) ->
     ConsoleLevel = conf_get("log.console_handler.level", Conf, undefined),
     FileLevels = [conf_get("level", SubConf) || {_, SubConf}
-                    <- maps:to_list(conf_get("log.file_handlers", Conf, #{}))],
+                    <- logger_file_handlers(Conf)],
     case FileLevels ++ [ConsoleLevel || ConsoleLevel =/= undefined] of
         [] -> warning; %% warning is the default level we should use
         Levels ->
             least_severe_log_level(Levels)
     end.
 
+logger_file_handlers(Conf) ->
+    Handlers = maps:to_list(conf_get("log.file_handlers", Conf, #{})),
+    lists:filter(fun({_Name, Opts}) ->
+                         B = conf_get("enable", Opts),
+                         true = is_boolean(B),
```

Comment:
Just curiosity: is this assert needed if the field type for `enable` is `boolean()`?  Could some other type be returned here?

Reply:
it's a an assertion to make sure future changes to a non-boolean will catch some attention